q2cli/tests/test_usage.py

```diff
@@ -197,3 +198,30 @@ def test_round_trip(action, example):
                        check=True,
                        cwd=tmpdir,
                        env={**os.environ})
+
+
+class ReplayResultCollectionTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.plugin = get_dummy_plugin()
+
+    def test_construct_and_access_collection(self):
+        action = self.plugin.actions['dict_of_ints']
+        use = CLIUsage()
+        action.examples['construct_and_access_collection'](use)
+        exp = """\
+## ignore: constructing result collection ##
+rc_name=rc-in/
+ext=.qza
+keys=( a b )
+names=( ints-a.qza ints-b.qza )
+construct_result_collection
+##
+qiime dummy-plugin dict-of-ints \\
+  --i-ints rc-in/ \\
+  --o-output rc-out/
+## ignore: accessing result collection member ##
+ln -s rc-out/b.qza ints-b-from-collection.qza
+##"""
+
+        self.assertEqual(exp, use.render())
```

Comment:
`build_header` and `build_footer` are methods that aren't called here

Reply:
Got it, that's part of the replay API, this will be something to keep in mind if collections get constructed in other non-replay usage examples. But we can revisit that in the future.